### PR TITLE
[Apache v2] Adding nodes 3/3 : add_child_comment() 

### PR DESCRIPTION
**Pull request by joohoi**
This PR implements add_child_comment()

Note: this PR uses temporarily #7497 as base to help to hide the unnecessary diffs. This also means that the tests won't run before it's changed back to apache-parser-v2

### CODE REVIEW

File: certbot-apache/certbot_apache/tests/augeasnode_test.py [hunk 1]
```diff
@@ -144,6 +144,16 @@ def test_set_parameters_delete(self):
                 found = True
         self.assertTrue(found)
 
+    def test_add_child_comment(self):
+        newc = self.config.parser_root.primary.add_child_comment("The content")
+        comments = self.config.parser_root.find_comments("The content")
+        self.assertEqual(len(comments), 1)
+        self.assertEqual(
+            newc.metadata["augeaspath"],
```
Comment (sydneyli) [inline certbot-apache/certbot_apache/tests/augeasnode_test.py]: Should this also be `primary.metadata`?

File: certbot-apache/certbot_apache/tests/augeasnode_test.py [hunk 2]
```diff
@@ -144,6 +144,16 @@ def test_set_parameters_delete(self):
                 found = True
         self.assertTrue(found)
 
+    def test_add_child_comment(self):
+        newc = self.config.parser_root.primary.add_child_comment("The content")
```
Comment (sydneyli) [inline certbot-apache/certbot_apache/tests/augeasnode_test.py]: Remove `primary`?

File: certbot-apache/certbot_apache/tests/augeasnode_test.py (hunk 1 above)
Comment (joohoi) [reply certbot-apache/certbot_apache/tests/augeasnode_test.py]: Good catch, and you are right. There's more than meets the eye though: All "read-only" operations towards instance variables already inherently are, as `DualParserNode` returns the value from primary from `__getattr__`.

I'm mainly accessing `primary` directly where I need to count the calls, and in all cases where I need to make "write" operations to not to do the writes twice as you already figured out :)

**Comment by joohoi** [comment]
This should be ready to merge

**Pull request merged**